Assets/BossRoom/Scripts/Client/UI/HeroActionBar.cs

```diff
@@ -106,9 +91,9 @@ void OnButtonClickedDown(int buttonIndex)
 
             switch (buttonIndex)
             {
-                case 0: m_InputSender.RequestAction(m_CharacterData.Skill1, SkillTriggerStyle.UI); break;
-                case 1: m_InputSender.RequestAction(m_CharacterData.Skill2, SkillTriggerStyle.UI); break;
-                case 2: m_InputSender.RequestAction(m_CharacterData.Skill3, SkillTriggerStyle.UI); break;
+                case 0: m_InputSender.RequestAction(m_IsOtherPlayerSelected ? ActionType.GeneralRevive : m_NetState.CharacterData.Skill1, SkillTriggerStyle.UI); break;
```

Comment:
Question: How could a new developer that takes a look at this, will be able to understand what those magic index values represent for Boss Room?

Reply:
I didn't write this code originally, but I found it fairy obvious how things were set up. If you're recommending we create an enum for the buttons instead of using an array, we can definitely do that! (But I don't think it would clarify the code for me.)

Attempted some simple clarifications:
- since the button array is hard-coded to support 4 buttons exactly, I added commentary saying so, and an Assert to confirm it at startup
- Rectored tertiary operator in case 0 switches to use a separate boolean to more explicitly convey intent

Reply:
I can't speak for @Cosmin-B, but I think he meant the indices used in the switch statement, which come from the UIHUDButton class. Would defining each Button out explicitly (instead of having an array) help in readability?

Reply:
I think they will, Fernando.

Reply:
ok, assigned constants for all hard-coded values in the class.

Reply:
oh, I should mention that I tried implementing it with the arrays completely removed, but it was ugly for two reasons:

- we need to keep track of each button's cached ref to their `UITooltipDetector` component. That's currently done by just having a second array with identical indices. Without the arrays, we need a different way to associate the two fields (e.g. a struct for each button+tooltip combo, or a switch-statement that returns the tooltip for a specific button). That's fine but it involves lots of ugly setup/teardown code.
- the reason those switch statements exist is because all buttons call the same callback events with different IDs. Either we need separate callback events for each button (eight total: four button-down and four button-up), or we have the same situation as we have now, where we need constants. 

So ultimately I think just using constants are the simplest solution here

Reply:
I do like this idea, a lot more, although the current implementation with constants, also looks way better than before.
`if this still seems bad, I'd next try an enum approach: using a private enum to index an array of structs, each struct containing the button, the tooltip, and any other associated data. (That's how I would probably have written this class if I'd made it originally.) That gives each button its own serialized variable, and hides the array internally. More code overall, but let me know if that is preferable.`

Reply:
Would you be able to do this now?

Reply:
Here is that version. As you can see it's a little bit more code. But one of the benefits is that we can remove the `ButtonID` int from `UIHUDButton`, which I think is a win.